packages/EasyBankFiles/src/Parsers/DirectEntry/Parser.php

```diff
@@ -75,67 +87,110 @@ public function getErrors(): array
+            $lineNumber++;
+        }
     }
 
-    /**
-     * {@inheritDoc}
-     */
     protected function processLine(int $lineNumber, string $line): void
     {
-        // code is the first character in line
         $code = $line[0] ?? self::EMPTY_LINE_CODE;
-        $lineLength = \strlen($line);
-
-        if ($code === self::HEADER && $lineLength >= self::MIN_HEADER_LINE_LENGTH) {
-            $this->header = $this->processHeader($line);
 
-            return;
+        if ($this->checkMinLength($line, $code)) {
+            $lineProcessingResult = match ($code) {
+                self::RECORD_TYPE_HEADER => $this->processHeader($line),
+                self::RECORD_TYPE_TRAILER => $this->processTrailer($line),
+                self::RECORD_TYPE_TRANSACTION_1 => $this->processTransactionType1($line),
+                self::RECORD_TYPE_TRANSACTION_2 => $this->processTransactionType2($line),
+                self::RECORD_TYPE_TRANSACTION_3 => $this->processTransactionType3($line),
+                default => $this->isFinalLine($line)
+            };
+
+            if ($lineProcessingResult === true) {
+                return;
+            }
         }
 
-        if ($code === self::TRAILER && $lineLength >= self::MIN_TRAILER_LINE_LENGTH) {
-            $this->trailer = $this->processTrailer($line);
+        $this->errors[] = new Error(\compact('line', 'lineNumber'));
+    }
 
-            return;
+    private function addTransactionToCurrentBatch(TransactionType1|TransactionType2|TransactionType3 $transaction): bool
+    {
+        if (isset($this->currentBatch) === false || $this->currentBatch->hasTrailer()) {
+            $this->currentBatch = null;
+
+            return false;
         }
+        $this->currentBatch->addTransaction($transaction);
 
-        if (
-            ($code === self::TRANSACTION_1 || $code === self::TRANSACTION_2) &&
-            $lineLength >= self::MIN_TRANSACTION_LINE_LENGTH
-        ) {
-            $this->transactions[] = $this->processTransaction($line);
+        return true;
+    }
 
-            return;
-        }
+    private function checkMinLength(string $line, string $code): bool
+    {
+        return isset(self::MIN_RECORD_LENGTH[$code]) === false || \strlen($line) >= self::MIN_RECORD_LENGTH[$code];
+    }
 
-        $this->errors[] = new Error(\compact('line', 'lineNumber'));
+    private function isFinalLine(string $line): bool
+    {
+        return (\trim($line) === self::FINAL_LINE || \trim($line, '9') === '') &&
+            ($this->currentBatch === null || $this->currentBatch->hasTrailer());
     }
 
     /**
-     * Process header block of line.
+     * @return array<mixed>
+     *
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
      */
-    private function processHeader(string $line): Header
+    private function parseCommonTransactionAttributes(string $line): array
+    {
+        /** @var string|false $accountName */
+        $accountName = \substr($line, 30, 32);
```

Comment:
Let's sort by second parameter (`offset`) in all methods.This will improve comparing with docs.

Reply:
I agreed, done